### PR TITLE
feat: support glob in `project` key

### PR DESCRIPTION
**Pull request by nix6839**
Fix: https://github.com/ota-meshi/eslint-plugin-astro/issues/342

Code from:
- https://github.com/typescript-eslint/typescript-eslint/blob/a14ba9d45891958962163c90615a0ab270e15a38/packages/typescript-estree/src/parseSettings/resolveProjectList.ts#L99-L101
- https://github.com/typescript-eslint/typescript-eslint/blob/a14ba9d45891958962163c90615a0ab270e15a38/packages/typescript-estree/src/parseSettings/getProjectConfigFiles.ts#L26-L34

### CODE REVIEW

File: src/parser/ts-for-v5/resolve-project-list.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/**
+ * Code from: https://github.com/typescript-eslint/typescript-eslint/blob/v5.62.0/packages/typescript-estree/src/parseSettings/resolveProjectList.ts
+ * Code updated to exclude caching.
+ */
+
+import { sync as globSync } from "globby";
+import isGlob from "is-glob";
+import * as path from "path";
+import * as ts from "typescript";
```
Comment (ota-meshi) [inline src/parser/ts-for-v5/resolve-project-list.ts:9]: Do you think users who don't have typescript installed will get an error on import?

We may need to load it dynamically inline instead.

`createRequire(path.join(process.cwd(), "__placeholder__.js"))("typescript")`

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Thank you for this PR! I have one thing to worry about.

File: src/parser/ts-for-v5/resolve-project-list.ts (hunk 1 above)
Comment (nix6839) [reply src/parser/ts-for-v5/resolve-project-list.ts:9]: Good catch, thank you! However, I have one question. What is `"__placeholder__.js"`?

Comment (ota-meshi) [reply src/parser/ts-for-v5/resolve-project-list.ts]: `"__placeholder__.js"` has no particular meaning. It is okay to use `index.js` or `foo.js` 😉

File: src/parser/ts-for-v5/resolve-project-list.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+/**
+ * Below code from: https://github.com/typescript-eslint/typescript-eslint/blob/v5.62.0/packages/typescript-estree/src/create-program/shared.ts#L53-L71
+ */
+
+// typescript doesn't provide a ts.sys implementation for browser environments
+const useCaseSensitiveFileNames =
+  ts.sys !== undefined ? ts.sys.useCaseSensitiveFileNames : true;
+const correctPathCasing = useCaseSensitiveFileNames
```
Comment (ota-meshi) [inline src/parser/ts-for-v5/resolve-project-list.ts:91]: I think we probably need to rewrite it like this:

```ts
let correctPathCasing = (filePath: string): string => {
  const ts: TS = createRequire(path.join(process.cwd(), "__placeholder__.js"))(
    "typescript",
  );
  const useCaseSensitiveFileNames = ts.sys !== undefined ? ts.sys.useCaseSensitiveFileNames : true;
  correctPathCasing = useCaseSensitiveFileNames
    ? (filePath: string): string => filePath
    : (filePath: string): string => filePath.toLowerCase();
  return correctPathCasing(filePath);
}
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: src/parser/ts-for-v5/resolve-project-list.ts (hunk 2 above)
Comment (nix6839) [reply src/parser/ts-for-v5/resolve-project-list.ts]: I made the unbelievable mistake of producing the same error in a different way. Thank you 😄

**Review by ota-meshi — APPROVED** [review]
LGTM! Thanks so much!

**Pull request merged**